client/request.go

```diff
@@ -83,7 +83,13 @@ func (cli *Client) sendClientRequest(ctx context.Context, method, path string, q
 	}
 
 	req, err := cli.newRequest(method, path, query, body, headers)
-	req.URL.Host = cli.addr
+	host := cli.addr
+	if cli.proto == "unix" || cli.proto == "npipe" {
+		// For local communications, it doesn't matter what the host is. We just
+		// need a valid and meaningful host name. (See #189)
+		host = "docker"
```

Comment:
shouldn't this set `req.Host`? We override the header (the thing that causes the problem on the server) only in the case that we need to talk to a Host that would cause a problem. The request still needs to go to the right place.